api/v1alpha08/sonataflowplatform_services_types.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha08
+
+// ServicesPlatformSpec describes the desired service configuration for "prod" workflows.
+type ServicesPlatformSpec struct {
+	// Deploys the Data Index service for use by "prod" profile workflows.
+	// +optional
+	DataIndex *ServicesPodTemplateSpec `json:"dataIndex,omitempty"`
+	// Persists service(s) to a datasource of choice. Ephemeral by default.
+	// +optional
+	Persistence *PersistenceOptions `json:"persistence,omitempty"`
+}
+
+// ServicesPodTemplateSpec describes the desired custom Kubernetes PodTemplate definition for the deployed flow.
+//
+// The ContainerSpec describes the container where the actual service is running. It will override any default definitions.
+// For example, to override the image one can use `.spec.services.dataIndex.container.image = my/image:tag`.
+type ServicesPodTemplateSpec struct {
+	// Container is the Kubernetes container where the service should run.
+	// One can change this attribute in order to override the defaults provided by the operator.
+	// +optional
+	Container *ContainerSpec `json:"container,omitempty"`
```

Comment:
This is a question rather than an objection:

f we look in FlowPodTemplateSpec, we have  Container ContainerSpec `json:"container,omitempty" **no pointers.** , and here we have Container *ContainerSpec. What's the benefit of having pointers now?
I think that, if we don't have strong reason for pointers, we should try to align our data structures.

Reply:
I think in, this case, we should remove the pointers to have access to the structure without nil pointers errors. Usually, we use pointers in the CRD to handle optional parameters.